Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Lib/DriverScore.hs

```diff
@@ -124,6 +128,35 @@ eventPayloadHandler _ DST.OnRideCompletion {..} = do
           pure . maybe 0 (const 1) $ (.nightShiftCharge) =<< mbFareParam
         Nothing -> pure 0
 
+updateDailyStats :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r) => Id DP.Person -> Id DMOC.MerchantOperatingCity -> DR.Ride -> m ()
+updateDailyStats driverId merchantOpCityId ride = do
+  transporterConfig <- CTCQ.findByMerchantOpCityId merchantOpCityId >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
+  localTime <- getLocalCurrentTime transporterConfig.timeDiffFromUtc
+  ds <- SQDS.findByDriverIdAndDate driverId (utctDay localTime)
+  dailyStatsOfDriver' <- case ds of
+    Nothing -> do
+      id <- generateGUIDText
+      pure
+        DDS.DailyStats
+          { id = id,
+            driverId = driverId,
+            totalEarnings = fromMaybe 0 ride.fare,
+            numRides = 1,
+            totalDistance = fromMaybe 0 ride.chargeableDistance,
+            merchantLocalDate = utctDay localTime
+          }
+    Just dailyStats ->
+      pure
+        DDS.DailyStats
+          { id = dailyStats.id,
+            driverId = driverId,
+            totalEarnings = dailyStats.totalEarnings + fromMaybe 0 ride.fare,
+            numRides = dailyStats.numRides + 1,
+            totalDistance = dailyStats.totalDistance + fromMaybe 0 ride.chargeableDistance,
+            merchantLocalDate = utctDay localTime
+          }
+  SQDS.upsertByDriverId dailyStatsOfDriver'
```

Comment:
You are already checking if ds for that day exist or not then why do it again in upsert query just do create or update in the case match above.